contracts/rewards/src/state.rs

```diff
@@ -0,0 +1,389 @@
+        );
+        assert!(loaded.is_ok());
+        assert!(loaded.unwrap().is_none());
+
+        // same event id but different contract address, should still return none
+        let loaded = store.load_event(
+            event.event_id.clone(),
+            Addr::unchecked("different contract"),
+        );
+        assert!(loaded.is_ok());
+        assert!(loaded.unwrap().is_none());
+
+        // different event id, but same contract address, should still return none
+        let loaded = store.load_event("some other event".into(), event.contract);
+        assert!(loaded.is_ok());
+        assert!(loaded.unwrap().is_none());
+    }
+
+    #[test]
+    fn save_and_load_epoch_tally() {
+        let mut mock_deps = mock_dependencies();
+        let mut store = RewardsStore {
+            storage: &mut mock_deps.storage,
+        };
+
+        let epoch_num = 10;
+        let contract = Addr::unchecked("some contract");
+        let rewards_rate = Uint256::from(100u128).try_into().unwrap();
+        let tally = EpochTally::new(epoch_num, contract.clone(), rewards_rate);
+
+        let res = store.save_epoch_tally(&tally);
+        assert!(res.is_ok());
+
+        // check that we load the tally that we just saved
+        let loaded = store.load_epoch_tally(contract.clone(), epoch_num);
+        assert!(loaded.is_ok());
+        assert!(loaded.as_ref().unwrap().is_some());
+        assert_eq!(loaded.unwrap().unwrap(), tally);
+
+        // different contract but same epoch should return none
+        let loaded = store.load_epoch_tally(Addr::unchecked("different contract"), epoch_num);
+        assert!(loaded.is_ok());
+        assert!(loaded.unwrap().is_none());
+
+        // different epoch but same contract should return none
+        let loaded = store.load_epoch_tally(contract.clone(), epoch_num + 1);
+        assert!(loaded.is_ok());
+        assert!(loaded.unwrap().is_none());
+
+        // different epoch and different contract should return none
+        let loaded = store.load_epoch_tally(contract.clone(), epoch_num + 1);
+        assert!(loaded.is_ok());
+        assert!(loaded.unwrap().is_none());
+    }
+
+    #[test]
+    fn save_and_load_rewards_pool() {
+        let mut mock_deps = mock_dependencies();
+        let mut store = RewardsStore {
+            storage: &mut mock_deps.storage,
+        };
+
+        let contract = Addr::unchecked("some contract");
+        let pool = RewardsPool::new(contract.clone());
+        let res = store.save_rewards_pool(&pool);
+        assert!(res.is_ok());
+
+        let loaded = store.load_rewards_pool(contract);
+
+        assert!(loaded.is_ok());
+        assert!(loaded.as_ref().unwrap().is_some());
+        assert_eq!(loaded.unwrap().unwrap(), pool);
+
+        let loaded = store.load_rewards_pool(Addr::unchecked("a different contract"));
+        assert!(loaded.is_ok());
+        assert!(loaded.as_ref().unwrap().is_none());
+    }
+}
```

Comment:
I don't really see any benefit in these tests, they just tie down the implementation. I'd rather wait and test at the high level functions of the reward contract

Reply:
yeah some of them are probably not necessary, but I think it's probably useful to test things like "same event id but different contract address loads a different event".